docs/concepts.rst

```diff
@@ -316,17 +316,42 @@ file as long as the ``major`` version matches.
     def message(greeting, subject):
         return f'{greeting} {subject}!!!'.upper()
 
+
 Be aware that Bionic can't detect every change that can affect your code's
-behavior.  It only looks at the code of the decorated function itself; if you
-change any other function or library that your decorated function depends on,
-Bionic won't notice.  Similarly, if your function is wrapped by a non-Bionic
-decorator, Bionic won't detect any code changes in that function at all.
-That's why this mode only provides a warning, rather than automatically
-invalidating the cache for you: to keep you in the habit of thinking carefully
-about versioning.
+behavior. Bionic will inspect the bytecode of each function, as well as any
+global variables and other functions it references [#bytecode_hash]_. However, it
+won’t detect changes of the following types:
+
+1. **Built-in or Installed Modules**: Bionic won’t detect changes in any built-in
+   modules or installed functions (modules accessed through Python’s
+   `installation paths <https://docs.python.org/3/library/sysconfig.html#installation-paths>`_).
+   Bionic *will* detect changes to modules in the same directory as your code,
+   and modules accessed through
+   `PYTHONPATH <https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH>`_.
+
+2. **Global Variables with Complex Types**: Bionic will only detect changes in
+   global variables that have simple types (int, string, bool, bytes, etc.), but
+   it will ignore changes for other types of variables.
+
+3. **Python Classes**: Bionic won’t detect changes to the code of Python classes.
+   [#classes_hash]_
+
+4. **Runtime Code**: Bionic doesn’t actually run any code when inspecting it, so
+   it won’t recursively inspect functions that are referenced dynamically. If a
+   function is not referenced straightforwardly (that is, as a global variable or
+   as an attribute of a global variable), Bionic may not inspect it. For example:
 
-However, if you do want to completely automate the versioning process, you
-can set Bionic to a "fully automatic" mode:
+   .. code-block:: python
+
+    def f():
+        a = get_a()  # We will inspect `get_a`.
+        b = module.submodule.get_b()  # We will inspect `get_b`.
+        import new_module
+        c = new_module.get_c()  # We won’t inspect `get_c`.
+        d = exec("get_d()")  # We definitely won’t inspect `get_d`!
+
+If you're not worried about false negatives, you can set Bionic to a "fully
```

Comment:
nit: I would say "*these* false negatives" to connect back to the previous section.